kubernetes/resource_kubernetes_daemonset.go

```diff
@@ -0,0 +1,281 @@
+							Type:        schema.TypeInt,
+							Description: "The number of old ReplicaSets to retain to allow rollback. This is a pointer to distinguish between explicit zero and not specified. Defaults to 10.",
+							Optional:    true,
+							Default:     10,
+						},
+						"selector": {
+							Type:        schema.TypeList,
+							Description: "A label query over pods that are managed by the DaemonSet.",
+							Optional:    true,
+							ForceNew:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: labelSelectorFields(true),
+							},
+						},
+						"strategy": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
+							Description: "The deployment strategy used to replace existing pods with new ones.",
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:        schema.TypeString,
+										Description: "Type of deployment. Can be 'RollingUpdate' or 'OnDelete'. Default is RollingUpdate.",
+										Optional:    true,
+										Default:     "RollingUpdate",
+									},
+									"rolling_update": {
+										Type:        schema.TypeList,
+										Description: "Rolling update config params. Present only if type = 'RollingUpdate'.",
+										Optional:    true,
+										Computed:    true,
+										MaxItems:    1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"max_unavailable": {
+													Type:        schema.TypeString,
+													Description: "The maximum number of DaemonSet pods that can be unavailable during the update. Value can be an absolute number (ex: 5) or a percentage of total number of DaemonSet pods at the start of the update (ex: 10%). Absolute number is calculated from percentage by rounding up. This cannot be 0. Default value is 1. Example: when this is set to 30%, at most 30% of the total number of nodes that should be running the daemon pod (i.e. status.desiredNumberScheduled) can have their pods stopped for an update at any given time. The update starts by stopping at most 30% of those DaemonSet pods and then brings up new DaemonSet pods in their place. Once the new pods are available, it then proceeds onto other DaemonSet pods, thus ensuring that at least 70% of original number of DaemonSet pods are available at all times during the update.",
+													Optional:    true,
+													Default:     1,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"template": {
+							Type:        schema.TypeList,
+							Description: "An object that describes the pod that will be created. The DaemonSet will create exactly one copy of this pod on every node that matches the template's node selector (or on every node if no node selector is specified). More info: https://kubernetes.io/docs/concepts/workloads/controllers/replicationcontroller#pod-template",
```

Comment:
This description should probably link to the DaemonSet specific 
https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#pod-template documentation or to the more general pod documentation https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/#pod-templates

Reply:
Done